### DISCUSSION TITLE
Where would I look if I wanted to implement a gone idle call on #0?

### DISCUSSION THREAD

**Discussion by NathanTech7713** [lisdude/toaststunt#88] [Q&A] [answered]
So I realise this is somewhat of a thing that maybe not many people would want.
And I'm definitely interested in learning to figure this out.

I just need some help on, the server would have the ability, pseudo code style to do something like:

server_loop starts:
 while shutdown is not called
    for connection in connections:
        process_any_commands
        if(connection->last_command_time > time+300):
          run_server-task(#0, gone_idle, x)

Is this something that would be doable? Where should I be looking here?

**Comment by Oliver2213** [comment, marked as answer]
Certainly doable, though I think you'd be better served just writing this in your database. Use a heart, task scheduler or similar mechanism if you have one. Schedule it to run every n, where n is your interval (300 seconds). Every run of your task, check connected players, and a property you add to your generic player class which keeps track of the last command time:
```MOO
for i in (connected_players())
  if(time() - `i.last_command_time ! E_PROPNF => time()' > 300)
    "Call your gone idle verb here";
  endif
  yin();
endfor
```
In #0:do_command, before you return, update that property on the player with the current time.
That code also falls back to the current time if the prop can't be found on the player, so the task won't break (or call) if you program and run it before you add the property.

**Reply by tvdijen** [reply]
```
if (idle_seconds(i) > 300)
```

No need for a property
